Marlin/src/feature/mmu/mmu2.cpp

```diff
@@ -508,9 +508,31 @@ static void mmu2_not_responding() {
 
     set_runout_valid(false);
 
-    if (index != extruder) {
+    if (index != extruder)
+    {
+      #if ENABLED(MMU_IR_UNLOAD_MOVE)
+        if (FILAMENT_PRESENT()) {
+          DEBUG_ECHOLNPGM("Unloading\n");
+          //filament_ramming(); //unloading from printer side
```

Comment:
If I understand correctly, `filament_ramming` is used when pulling filament from the nozzle, to produce a good shape to work with the MMU.

Why would you only do this when loading to nozzle, and not during tool change? It seems that tool change should have to unload filament from the nozzle also.

If it is ok to use filament_ramming in both cases, perhaps you could move this into `MMU2::unload()`. This also had the advantage that it would apply to the Unload command also.

Reply:
to explain this, there are two sources of ramming sequences for unloading

One is defined by the Slicer (at the moment only Prusa Slicer/Super Slicer) which is converted into Gcode before executing the T# command from an actual print
![image](https://user-images.githubusercontent.com/2419987/99698105-5bb13580-2acb-11eb-91cd-faf31c540c2c.png)

and the Second is the internal ramming sequence from the printer when using the LCD since there is no source of Gcode to instruct it
![image](https://user-images.githubusercontent.com/2419987/99698215-7be0f480-2acb-11eb-9fbf-31a0cac637c4.png)

load filament to nozzle () is a function called only when operating the LCD screen but not related or called during any of the printer-to-mmu protocols. It can only rely on filament_ramming built in to form the tip but isn't ideal in an actual long print and less reliable with different types of materials that's why prusa slicer has filament ramming as part of the settings which varies per filament setting.

It's the tool_change() function that loads the filament to the gears during printing as called by the T# command. By principle, the tool_change() is a less instruction-full function for the printer since the preceding and succeeding instructions to the printer is generated by the slicer but it coordinates with the MMU unit as to tell which filament # I want at the moment. This is where the slicer-generated-filament ramming gcode is inserted before the T# command, and tool change loading command gcode after.

Reply:
Thank you for this description. It may be worth adding come comments to the code describing some of these reasons. They may not be obvious to new people in the code why _some_ unloads perform ramming and others do not.